url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwin/initialization/Full.java

```diff
@@ -0,0 +1,56 @@
+package darwin.initialization;
+
+import darwin.geneticOperation.UnableToPerformOperationException;
+import darwin.nodeFilter.ReturnType;
+import darwin.nodeFilter.Terminals;
+import darwin.nodeFilter.Functions;
+import darwin.population.Node;
+import darwin.population.TypeMismatchException;
+import darwin.problemObject.NodeFactory;
+
+/**
+ * Generator representing the Full method, as described by Koza
+ * @author Kevin Dolan
+ */
+public class Full implements Generator {
+
+	private NodeFactory nodeFactory;
+	
+	public Full(NodeFactory nodeFactory) {
+		this.nodeFactory = nodeFactory;
+	}
+	
+	@Override
+	public Node generate(int depth, int size, Class<?> returnType) throws UnableToPerformOperationException {
+		Node current;
+		
+		try {
+			if(depth > 1 && size > 1) 
+				current = nodeFactory.getNode(new Functions(returnType));
+			else 
+				current = nodeFactory.getNode(new Terminals(returnType));
+		}
+		catch(NoAvailableNodeException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 36.